utils/utils.go

```diff
@@ -76,14 +77,33 @@ func GetAbsolutePath(path string) string {
 }
 
 func NameFromRepository(repository string) string {
-	if strings.HasPrefix(repository, "https://") {
-		return strings.Replace(strings.Split(strings.TrimPrefix(repository, "https://"), "/")[2], ".git", "", 1)
-	} else if strings.HasPrefix(repository, "git@") {
-		return strings.Replace(strings.Split(repository, "/")[1], ".git", "", 1)
-	} else {
-		logger.Fatal("Unable to get name from repository: %s", repository)
-	}
-	return ""
+	name := repository
+
+	scheme := regexp.MustCompile("^[a-zA-Z]+://")
+	needle := scheme.FindStringIndex(name)
+	if needle != nil {
+		name = name[needle[1]:]
+	}
+	user := regexp.MustCompile("^[a-zA-Z0-9_.-]+@")
```

Comment:
I think thje character set in the regex should also include `:` for http urls including username & password, or am I wrong?

Reply:
Yes, you're right. I added it.

Reply:
Thanks a lot!